lib/versioning/gradle/index.ts

```diff
@@ -0,0 +1,210 @@
+import { NewValueConfig, VersioningApi } from '../common';
+import {
+  compare,
+  isValid,
+  isVersion,
+  parseMavenBasedRange,
+  parsePrefixRange,
+  RangeBound,
+  tokenize,
+  TokenType,
+} from './compare';
+
+export const id = 'gradle';
+export const displayName = 'Gradle';
+export const urls = [
+  'https://docs.gradle.org/current/userguide/single_versions.html#version_ordering',
+];
+export const supportsRanges = true;
+export const supportedRangeStrategies = ['pin'];
+
+const equals = (a: string, b: string): boolean => compare(a, b) === 0;
+
+const getMajor = (version: string): number | null => {
+  if (isVersion(version)) {
+    const tokens = tokenize(version.replace(/^v/i, ''));
+    const majorToken = tokens[0];
+    if (majorToken && majorToken.type === TokenType.Number) {
+      return +majorToken.val;
+    }
+  }
+  return null;
+};
+
+const getMinor = (version: string): number | null => {
+  if (isVersion(version)) {
+    const tokens = tokenize(version.replace(/^v/i, ''));
+    const majorToken = tokens[0];
+    const minorToken = tokens[1];
+    if (
+      majorToken &&
+      majorToken.type === TokenType.Number &&
+      minorToken &&
+      minorToken.type === TokenType.Number
+    ) {
+      return +minorToken.val;
+    }
+    return 0;
+  }
+  return null;
+};
+
+const getPatch = (version: string): number | null => {
+  if (isVersion(version)) {
+    const tokens = tokenize(version.replace(/^v/i, ''));
+    const majorToken = tokens[0];
+    const minorToken = tokens[1];
+    const patchToken = tokens[2];
+    if (
+      majorToken &&
+      majorToken.type === TokenType.Number &&
+      minorToken &&
+      minorToken.type === TokenType.Number &&
+      patchToken &&
+      patchToken.type === TokenType.Number
+    ) {
+      return +patchToken.val;
+    }
+    return 0;
+  }
+  return null;
+};
+
+const isGreaterThan = (a: string, b: string): boolean => compare(a, b) === 1;
+
+const unstable = new Set([
```

Comment:
These are common specifiers brought from Maven versioning (+ additional one for `mt`).